app/employee-app/backend/modules/database/db_functions.bal

```diff
@@ -78,3 +79,107 @@ public isolated function getEmployees(EmployeeFilter filters, int 'limit, int of
+# + return - Returns the list of business units
+public isolated function getOrgStructure(string[]? employeeStatuses) returns OrgDataResponse|error {
+
+    OrgItem[] orgItems = [];
+    FlatList flatList = {
+        buFlatList: [],
+        teamFlatList: [],
+        unitFlatList: []
+    };
+    stream<OrgDataDB, sql:Error?> resultStream = databaseClient->query(getOrgStructureQuery(employeeStatuses));
+    error? orgResult = from OrgDataDB orgData in resultStream
+        do {
+            Team[]|error teams = orgData.children.fromJsonStringWithType();
+            if teams is error {
+                return error(
+                    string `An error occurred when retrieving teams data of ${orgData.name}!`, teams
+                );
+            }
+
+            flatList = processFlatList(flatList, orgData.name, 0, teams);
+            orgItems.push(processOrgHierarchy(orgData.name, 0, teams));
+        };
+
+    if orgResult is sql:Error {
+        return error("An error occurred when retrieving organization details!");
+    }
+    stream<LocationData, error?> distinctLocations =
+            databaseClient->query(getDistinctEmployeeLocationQuery(employeeStatuses));
+    string[]? locationResults = check from LocationData locationResult in distinctLocations
+        select locationResult.location;
+
+    return {orgStructure: orgItems, flatList, locations: locationResults ?: []};
+}
+
+# Recursive method to build the flatList.
+#
+# + flatList - Initialized flatList 
+# + name - Name of each business unit/team/unit
+# + level - Level number 
+# + children - Children of each business unit/team/unit
+# + return - Returns the updated flatlist
+public isolated function processFlatList(
+        FlatList flatList,
+        string name,
+        int level,
+        OrgType[]? children
+) returns FlatList {
+
+    if level == 0 && flatList.buFlatList.indexOf(name) is () {
+        flatList.buFlatList.push(name);
+    } else if level == 1 && flatList.teamFlatList.indexOf(name) is () {
+        flatList.teamFlatList.push(name);
+    } else if level == 2 && flatList.unitFlatList.indexOf(name) is () {
+        flatList.unitFlatList.push(name);
+    }
+
+    if children is OrgType[] {
+        foreach OrgType child in children {
+            if child is Team {
+                _ = processFlatList(flatList, child.name, level + 1, child.children);
+            } else {
+                _ = processFlatList(flatList, child.name, level + 1, []);
+            }
+        }
+    }
+
+    return flatList;
+}
+
+# Recursive method to build the organization hierarchy.
+#
+# + name - Name of the business unit/team/unit 
+# + children - Children of each business unit/team/unit
+# + level - Level number 
+# + return - Returns organization item with name, level, type, type name and its children
+public isolated function processOrgHierarchy(
+        string name,
+        int level,
+        OrgType[]? children)
+returns OrgItem {
```

Comment:
This indentation seems bit odd. Shall we have a chat @jsdjayanga @shayanmalinda @yuk7hi @avishka

Reply:
Updated. But this format is used when arguments exceed 120 characters.
For example,
```
    resource function get test(
            http:RequestContext ctx,
            http:Request req,
            string? email = (),
            string? startDate = (),
            string? endDate = (),
            boolean? isActive = (),
            int? 'limit = (),
            int? offset = 0
    ) returns TestRecord|http:Forbidden|http:InternalServerError|http:BadRequest {
```